### PR TITLE
PARQUET-1966: Fix build with JDK11 for JDK8

### PR DESCRIPTION
**Pull request by gszadovszky**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-XXX
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Comment by dossett** [comment]
That's a nice use of profiles.  I'm (non-binding) +1. I don't see any logs for the travis build failures, don't know if they've expired or maybe the tests just failed to launch for some reason.

**Comment by gszadovszky** [comment]
Thanks, @dossett. I am not sure about the Travis status either. I've restarted the build, let's see what happens.

**Comment by gszadovszky** [comment]
Close/re-open to trigger tests.
@shangxinli, if you have time could you please review this? The release is blocked on this one.

**Review by wangyum — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -393,6 +393,7 @@
         <!-- Override source and target from the ASF parent -->
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.1</version>
```
Comment (Fokko) [inline pom.xml:396]: We could also bump the parent POM to 22: https://github.com/apache/maven-apache-parent/blob/apache-22/pom.xml

**Comment by shangxinli** [comment]
For the failed test "testMemoryManagerUpperLimit", I am not sure it is caused by this change, or the test itself is unstable.

File: pom.xml (hunk 1 above)
Comment (gszadovszky) [reply pom.xml:396]: I would not update it for now (just before the release). I think, it is better to bump such versions for the next release. The bump of the `maven-compiler-plugin` was required to support the `release` config.

**Comment by gszadovszky** [comment]
> For the failed test "testMemoryManagerUpperLimit", I am not sure it is caused by this change, or the test itself is unstable. It seems the failure is because the pool size is larger than expected.
> 
> "should be within 10% of the expected value (expected = 453745044 actual = 505046624)"

Yes, the test is flaky. Based on git history we've had to increase the percentage 3 times already. (I am not sure if this test make sense this way but I did not want to drop it for now.) Meanwhile, I have not seen this test fail until the transition to the github actions. I guess there are some environmental differences between the new actions and Travis.

**Comment by shangxinli** [comment]
> > For the failed test "testMemoryManagerUpperLimit", I am not sure it is caused by this change, or the test itself is unstable. It seems the failure is because the pool size is larger than expected.
> > "should be within 10% of the expected value (expected = 453745044 actual = 505046624)"
> 
> Yes, the test is flaky. Based on git history we've had to increase the percentage 3 times already. (I am not sure if this test make sense this way but I did not want to drop it for now.) Meanwhile, I have not seen this test fail until the transition to the github actions. I guess there are some environmental differences between the new actions and Travis.
> 
> But this change is not related to the flaky test. It is related to the issue caused by the first RC built with jdk11 and failed in a jre8 runtime.

Sounds good

**Review by shangxinli — APPROVED** [review]

**Pull request merged**